dataflow/src/lib.rs

```diff
@@ -100,6 +103,7 @@ pub fn shard_by(dt: &core::DataType, shards: usize) -> usize {
             hasher.write(s.as_bytes());
             hasher.finish() as usize % shards
         }
+        core::DataType::None => thread_rng().gen_range(0, shards),
```

Comment:
Creating a new rng every time here is a little sad. As is doing random assignment. Can we do round-robin instead? I don't recall how many places this is called from, but I think it should be doable to have the sharder/mutator remember where it sent the last one.

Reply:
Changed to round-robin in the last commit. Required a few extra struct variables and mutations though. Might be nicer to make something like a `DomainSharder` and just handle it internally there? That could result in more allocations in places such as the backlog though. 

Perhaps split it into two? `DomainSharder::shard_by` and `shard_by`?